include/swift/Sema/IDETypeChecking.h

```diff
@@ -40,6 +40,10 @@ namespace swift {
   /// \returns true on convertible, false on not.
   bool isConvertibleTo(Type T1, Type T2, DeclContext &DC);
 
+  /// \brief Determine whether a declaration is implicitly convertible to a
+  /// given type. Unlike isConvertibleTo, this handles generic types correctly.
+  bool isDeclConvertibleTo(const ValueDecl *decl, Type fromType, Type toType);
```

Comment:
Can you elaborate on what the 'appropriate substitutions' would be, and how they would let us match up a member function type with a nonmember expected function type?